qubes_menu/appmenu.py

```diff
@@ -252,13 +252,18 @@ def _key_press(self, _widget, event):
         """
         Keypress handler, to allow closing the menu with an ESC key and to fix
         some issues with space (as we have search by default, we should not
-        react to space with launching an app).
+        react to space with launching an app, but instead put the space
+        into the search bar).
         """
         if event.keyval == Gdk.KEY_Escape:
             self.hide_menu()
         if event.keyval == Gdk.KEY_space:
-            if not isinstance(self.get_active_window().get_focus(),
-                              Gtk.SearchEntry):
+            current_widget = self.get_active_window().get_focus()
+            if isinstance(current_widget,
+                          Gtk.SearchEntry):
+                p = current_widget.get_position()
+                current_widget.set_text(current_widget.get_text() + " ")
```

Comment:
Now when I type space in the middle of search entry (move cursor with arrows first), then the space doesn't appear there, only at the end...

Reply:
FIXED IT FINALLY, no more raging consumption of spaces or weird space behavior